src/hmem_ze.c

```diff
@@ -403,26 +407,36 @@ static int ze_hmem_init_fds(void)
 		if (ret < 0 || ret >= NAME_MAX)
 			goto err;
 
-		dev_fds[i] = open(dev_name, O_RDWR);
-		if (dev_fds[i] == -1)
+		dev_fds[num_pci_devices] = open(dev_name, O_RDWR);
+		if (dev_fds[num_pci_devices] == -1)
 			goto err;
-		i++;
+		uuid_str = strtok(ent->d_name, ":");
+		uuid_str = strtok(NULL, ":");
+		uuid = (uint8_t) strtol(uuid_str, NULL, 16);
+
+		for (i = 0; i < num_devices; i++) {
+			if (dev_info[i].uuid.id[8] == uuid)
+				dev_info[i].pci_device = num_pci_devices;
```

Comment:
For the full pci address `domain:bus:dev.func`, the `domain` part needs to be considered as well. For example, here is a list of pci addresses on an actual machine with 6 PVCs: `0000:18:00.0`, `0000:42:00.0`, `0000:6c:00.0`, `0001:18:00.0`, `0001:42:00.0`, `0001:6c:00.0`.

Reply:
@j-xiong Updated. Do the dev and func need to be considered as well?

Reply:
So far dev and func are always 0 and can be ignored.